utils/relay-manager.js

```diff
@@ -29,43 +52,166 @@ class RelayManager {
       methodData,
       chainId,
       nonce,
-      _gasPrice,
-      _gasLimit,
+      gasPrice,
+      gasLimit,
       _refundToken,
       _refundAddress,
     );
 
-    if (_estimate === true) {
-      const gasUsed = await this.relayerManager.estimate.execute(
-        _wallet.address,
-        _module.address,
-        methodData,
-        nonce,
-        signatures,
-        _gasPrice,
-        _gasLimit,
-        _refundToken,
-        _refundAddress,
-        { gas: _gasLimitRelay, gasPrice: _gasPrice },
-      );
-      return gasUsed;
-    }
+    const executeData = this.relayerManager.contract.methods.execute(
+      _wallet.address,
+      _module.address,
+      methodData,
+      nonce,
+      signatures,
+      gasPrice,
+      gasLimit,
+      _refundToken,
+      _refundAddress).encodeABI();
+
+    // Get the number of zero and non-zero bytes in the relayer.execute calldata
+    const nonZerosString = executeData.toString().slice(2).replace(/0/g, "");
```

Comment:
This regex will remove all zero characters but we want to only remove zero bytes (i.e. `00` at even positions). We could use that regex instead:
```
const nonZerosString = executeData.toString().slice(2).replace(/00(?=(..)*$)/g,"")
```

Reply:
Good find!